Created DSL where stubs and expectations can be set

bbalser opened the pull request:
Created DSL where stubs and expectations can be set with a syntax that looks like a normal function call.
```elixir
test "invokes add once and mult twice" do
  Calculator
  |> stub(:add, fn x, y -> :stub end)
  |> expect(:add, fn x, y -> x + y end)
  |> expect(:mult, 2, fn x, y -> x * y end)

  assert Calculator.add(2, 3) == 5
  assert Calculator.mult(2, 3) == 6

  assert Calculator.add(2, 3) == :stub
end
```
could now be expressed as
```elixir
  test "basic example" do
    stub Calculator.add(_x, _y), do: :stub
    expect Calculator.add(x, y), do: x + y
    expect Calculator.mult(x, y), do: x * y

    assert Calculator.add(2, 3) == 5
    assert Calculator.mult(2, 3) == 6

    assert Calculator.add(2, 3) == :stub
  end
```
Hope you like it.

edgurgel commented on lib/mimic/dsl.ex:
```diff
@@ -0,0 +1,92 @@
+  @doc false
+  defmacro __using__(_opts) do
+    quote do
+      import Mimic, except: [allow: 3, expect: 3, except: 4]
```
Should this be `expect: 4` ?

edgurgel commented:
Thanks for submitting this change 🎉 

I really like that this is optional! Should we also add a section to the README.md so people can see that this DSL is available?

calvin-kargo commented:
Cool PR !! definitely is going to try this out when it's released!

A bit wondering though, why on `stub` on DSL mode is called `allow` ?

bbalser commented:
Changed `allow` to `stub`, because I wasn't thinking clearly when I wrote that ;)

Added `num_calls` option to expect calls

Added section to README

edgurgel commented:
This is great! thank you!
